File: src/2025/Arrays/49.rs
use std::collections::HashMap;

impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut keys: HashMap<String, Vec<String>> = HashMap::new();
        for word in strs.iter() {
            let mut chars: Vec<char> = word.chars().collect();
            chars.sort();
            let key: String = chars.into_iter().collect();
            keys.entry(key).or_insert(vec![]).push(word.clone());
        }

        keys.into_values().collect()
    }
}

File: src/2025/XPrep/49.cc
class Solution {
  public:
    vector<vector<string>> groupAnagrams(vector<string> &strs) {
        unordered_map<string, vector<string>> mp;
        for (string &str : strs) {
            string key = str;
            sort(key.begin(), key.end());
            mp[key].push_back(str);
        }

        vector<vector<string>> result;
        for (auto &kv : mp) {
            vector<string> grams;
            for (auto &word : kv.second) {
                grams.push_back(word);
            }
            result.push_back(grams);
        }
        return result;
    }
};